js/models/messages.js

```diff
@@ -1196,6 +1216,75 @@
       return errors[0][0];
     },
 
+    async handleMessageSentSuccess(sentMessage) {
+      const sentTo = this.get('sent_to') || [];
+
+      const isOurDevice = window.libsession.Protocols.MultiDeviceProtocol.isOurDevice(
+        sentMessage.device
+      );
+
+      // Handle the sync logic here
+      if (!isOurDevice && !this.get('synced') && !this.get('sentSync')) {
+        const contentDecoded = textsecure.protobuf.Content.decode(
+          sentMessage.plainTextBuffer
+        );
+        const { dataMessage } = contentDecoded;
+        this.sendSyncMessageOnly(dataMessage);
+
+        this.set({ sentSync: true });
```

Comment:
This can probably be placed in `sendSyncMessage` to make it easier and we can return early from the function if we already sent a sync message.